pageserver/src/tenant/timeline.rs

```diff
@@ -4465,43 +4465,29 @@ impl Timeline {
     ) -> anyhow::Result<()> {
         let mut guard = self.layers.write().await;
 
-        let mut duplicated_layers = HashSet::new();
-
-        let mut insert_layers = Vec::with_capacity(new_deltas.len());
-
         for l in new_deltas {
             if guard.contains(l.as_ref()) {
-                // expected in tests
-                tracing::error!(layer=%l, "duplicated L1 layer");
-
-                // good ways to cause a duplicate: we repeatedly error after taking the writelock
-                // `guard`  on self.layers. as of writing this, there are no error returns except
-                // for compact_level0_phase1 creating an L0, which does not happen in practice
-                // because we have not implemented L0 => L0 compaction.
-                duplicated_layers.insert(l.layer_desc().key());
+                // we should abort
+                error!("duplicate L1 layer; \
+                 likely we already overwrote the exiting file on disk with a possibly different bit pattern, \
+                 causing incoherency with in-memory copies (PS page cache) {l}");
+                std::process::abort(); // scary to run `struct Layer` desctructor (?)
             } else if LayerMap::is_l0(l.layer_desc()) {
-                bail!("compaction generates a L0 layer file as output, which will cause infinite compaction.");
-            } else {
-                insert_layers.push(l.clone());
+                // we should abort so we don't  otherwise we might leave some
+                error!("compaction generates a L0 layer file as output, which will cause infinite compaction.");
+                std::process::abort(); // if we returned an error, we'd not check the remaining `new_delta` layers for duplicates
```

Comment:
This is not needed either. It would surely be a restart loop every 20s.

Reply:
discussed offline